addons/account/models/account_move.py

```diff
@@ -748,7 +748,14 @@ def _compute_type_name(self):
 
     @api.depends('line_ids.account_id.account_type')
     def _compute_always_tax_exigible(self):
-        for record in self:
+        # prewarming cache with fields fetched in 'is_invoice'
+        # and '_collect_tax_cash_basis_values' to remove the
+        # need to fetch values for each move and move line and
+        # instead fetch them in bulk
+        moves = self.with_context(prefetch_fields=False).browse(self.ids)
+        moves._read(['move_type'])
+        moves.line_ids._read(['account_type', 'tax_line_id'])
+        for record in moves:
```

Comment:
```suggestion
        for record in self.with_context(prefetch_fields=False):
```
You are leaking the abstraction of the methods `is_invoice` and `_collect_tax_cash_basis_values`, meaning you are trying to pre-read fields used in those methods, but actually you don't know the content (as they can be overriden in 3rd party code, or in the future their implementation may change in standard code). And if you wanted to prewarm the cache to avoid a prefetch being triggered upon cache miss, you need to read *all* fields, you've missed quite a few for `line_ids` in `_collect_tax_cash_basis_values` (`balance`, `currency_id`, `amount_residual`, and others...)

The only reason the patch works as expected, is that you've disabled the prefetcher in the beginning for the `moves`. The context is also forwarded to the recordset that will be created when reading `line_ids` later on, so the reading of fields on it also will have the prefetcher disabled.

With that in mind:
- You can just disable the prefetcher for the loop as I've suggested
- Since we don't need to do the explicit `_read` as they are not useful here, we remove them, and the comment becomes unnecessary. If later on someone wants to know why is there a `prefetch_fields=False` on the loop, they can read the blame. 

Feel free to tryout my proposition, it should yield the same results as your patch, but is only 1 line diff.

Reply:
@pivi-odoo, you're right and it yields the same result. I have added the changes!